src/pairwise_dist/_pairwise_dist.pyx

```diff
@@ -0,0 +1,46 @@
+# cython: boundscheck=False
```

Comment:
There are others checks that you can remove for efficiency but I am sure that you know this already. :)